From cc0828da29065ff6cadc1b9cd5fc38dd1466b395 Mon Sep 17 00:00:00 2001
From: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Date: Thu, 22 Jun 2017 17:41:37 +0300
Subject: [PATCH 2132/5242] MLK-15048-2: video: Add proper config dependencies

commit  6317e8f1241a0ee46f9806ea6ca53065159620b9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently, if setting MXC_IPU to 'no' leads to a chain of build crashes.
This happens due to lack of proper config dependencies.

This patch fixes the following interdependencies:
	- framebuffer sync panel (FB_MXC_SYNC_PANEL)
		- should depend on IPU v3 (MXC_IPU_V3)
	- ADV7535 MIPI-to-HDMI converter (FB_MXC_ADV7535)
		- should depend on MIPI DSI (FB_MXC_MIPI_DSI)
	- framebuffer MXS LCD controller (FB_MXS)
		- should depend on framebuffer sync panel (FB_MXC_SYNC_PANEL)
	- camera v4l drivers + TV decoders that use IOCTL interface
	  (MXC_CAMERA_OV5640, MXC_CAMERA_OV5642, MXC_CAMERA_OV5640_MIPI,
	  MXC_TVIN_ADV7180)
		- should depend on V4L2 internal IOCTL interface
		  (VIDEO_V4L2_MXC_INT_DEVICE)

Signed-off-by: Cristina Ciocan <cristina-mihaela.ciocan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/mxc/capture/Kconfig |    4 ++++
 drivers/video/fbdev/mxc/Kconfig            |    1 +
 2 files changed, 5 insertions(+)

diff --git a/drivers/media/platform/mxc/capture/Kconfig b/drivers/media/platform/mxc/capture/Kconfig
index 6671b69..2aa4e15 100644
--- a/drivers/media/platform/mxc/capture/Kconfig
+++ b/drivers/media/platform/mxc/capture/Kconfig
@@ -31,6 +31,7 @@ config VIDEO_MXC_IPU_CAMERA
 config MXC_CAMERA_OV5640
         tristate "OmniVision ov5640 camera support"
         depends on !VIDEO_MXC_EMMA_CAMERA && I2C
+	depends on VIDEO_V4L2_MXC_INT_DEVICE
         ---help---
           If you plan to use the ov5640 Camera with your MXC system, say Y here.
 
@@ -43,12 +44,14 @@ config MXC_CAMERA_OV5640_V2
 config MXC_CAMERA_OV5642
 	tristate "OmniVision ov5642 camera support"
 	depends on !VIDEO_MXC_EMMA_CAMERA && I2C
+	depends on VIDEO_V4L2_MXC_INT_DEVICE
 	---help---
 	  If you plan to use the ov5642 Camera with your MXC system, say Y here.
 
 config MXC_CAMERA_OV5640_MIPI
 	tristate "OmniVision ov5640 camera support using mipi"
 	depends on !VIDEO_MXC_EMMA_CAMERA && I2C && MXC_MIPI_CSI2
+	depends on VIDEO_V4L2_MXC_INT_DEVICE
 	---help---
 	  If you plan to use the ov5640 Camera with mipi interface in your MXC system, say Y here.
 
@@ -67,6 +70,7 @@ config MXC_CAMERA_OV5647_MIPI
 config MXC_TVIN_ADV7180
 	tristate "Analog Device adv7180 TV Decoder Input support"
 	depends on !VIDEO_MXC_EMMA_CAMERA && I2C
+	depends on VIDEO_V4L2_MXC_INT_DEVICE
 	---help---
 	  If you plan to use the adv7180 video decoder with your MXC system, say Y here.
 
diff --git a/drivers/video/fbdev/mxc/Kconfig b/drivers/video/fbdev/mxc/Kconfig
index d5c439b..6f6589e 100644
--- a/drivers/video/fbdev/mxc/Kconfig
+++ b/drivers/video/fbdev/mxc/Kconfig
@@ -89,6 +89,7 @@ config FB_MXC_DCIC
 config FB_MXC_ADV7535
 	tristate "ADI ADV7535 support"
 	depends on I2C
+	depends on FB_MXC_MIPI_DSI
 	help
 	  Driver support for the ADV7535 DSI-to-HDMI module
 
-- 
1.7.9.5

